arch/arm64/core/mmu.c

```diff
@@ -997,7 +997,7 @@ size_t arch_virt_region_align(uintptr_t phys, size_t size)
 {
 	size_t alignment = CONFIG_MMU_PAGE_SIZE;
 	size_t level_size;
-	int level;
+	unsigned int level;
```

Comment:
I don't think that this is safe as it looks like BASE_XLAT_LEVEL can be 0. The following loop might never end in that case.